apps/mobile/app/src/main/java/com/glycemicgpt/mobile/plugin/PluginRegistry.kt

```
@@ -201,6 +225,126 @@ class PluginRegistry @Inject constructor(
+        if (pluginId in activePlugins) {
+            deactivatePluginInternal(pluginId).onFailure { e ->
+                Timber.w(e, "Failed to deactivate plugin %s during removal, continuing", pluginId)
+            }
+        }
+
+        // Shutdown and remove from registry
+        val plugin = plugins.remove(pluginId)
+        if (plugin != null) {
+            try {
+                plugin.shutdown()
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)